File: catalog/complete-workflows/research-summary.ee.js
/* global exports */

exports.summarize = function (values) {
  var clean = values.map(Number);
  var total = clean.reduce(function (sum, value) { return sum + value; }, 0);
  return {
    count: clean.length,
    mean: total / clean.length,
    minimum: Math.min.apply(null, clean),
    maximum: Math.max.apply(null, clean)
  };
};

File: catalog/complete-workflows/research-summary.ee.py
__all__ = ["summarize"]


def summarize(values):
    """Return portable summary values to the JavaScript entrypoint."""
    clean = [float(value) for value in values]
    return {
        "count": len(clean),
        "mean": sum(clean) / len(clean),
        "minimum": min(clean),
        "maximum": max(clean),
    }
